jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -215,12 +215,23 @@
 			accessMode = lookupMode;
 		}
 		
+		/* For Java 15, the default is not to check for the "java.lang.invoke" package.
+		 * For earlier releases, these lookups are illegal
+		  */
+		private static boolean lookupJLIPackageCheckDefault() {
+			/*[IF Java15]
+			return false;
+			/*[ELSE] Java15*/
+			return true;
+			/*[ENDIF] Java15*/
+		}
+		
 		Lookup(Class<?> lookupClass, Class<?> prevLookupClass, int lookupMode) {
-			this(lookupClass, prevLookupClass, lookupMode, true);
+			this(lookupClass, prevLookupClass, lookupMode, lookupJLIPackageCheckDefault());
 		}
 		
 		Lookup(Class<?> lookupClass, int lookupMode, boolean doCheck) {
-			this(lookupClass, null, lookupMode, doCheck);
+			this(lookupClass, null, lookupMode, lookupJLIPackageCheckDefault());
```

Comment:
This doesn't look right as it discards `doCheck`.  It should be
```suggestion
			this(lookupClass, null, lookupMode, doCheck);
```

It's up to the calling code to pass the right default or not supply one by using a different ctor.

Reply:
Modified the helper method as following:
```
		private static boolean lookupJLIPackageCheckDefault(boolean doCheck) {
			/*[IF Java15]
			return false;
			/*[ELSE] Java15*/
			return doCheck;
			/*[ENDIF] Java15*/
		}
```
This is to skip the package check in JDK15 regardless of incoming `doCheck` flag while keep same behaviours for pre-JDK15 levels.
Here it invokes `lookupJLIPackageCheckDefault(doCheck)` instead.